File: modules/common/src/main/java/com/hig/exceptions/NoAuthenticationException.java
package com.hig.exceptions;

import com.hig.result.type.CommonResult;
import com.hig.result.Result;
import com.hig.utils.MessageUtil;
import lombok.Getter;

/**
 * 인증에 실패할 경우 throw
 */
public class NoAuthenticationException extends BaseException {

	private static final long serialVersionUID = 6940889666952870454L;

	@Getter
	private final Result result;
	
	@Getter
	private final String message;
	
	public NoAuthenticationException() {
		super(CommonResult.AUTHENTICATION_FAIL);
		this.result = CommonResult.AUTHENTICATION_FAIL;
		this.message = CommonResult.AUTHENTICATION_FAIL.getMessage();
	}
	
	public NoAuthenticationException(String message) {
		super(message);
		this.result = CommonResult.AUTHENTICATION_FAIL;
		this.message = message;
	}
	
	public NoAuthenticationException(String message, Throwable cause) {
		super(message, cause);
		this.result = CommonResult.AUTHENTICATION_FAIL;
		this.message = message;
	}
	
	public NoAuthenticationException(Throwable cause) {
		super(cause);
		this.result = CommonResult.AUTHENTICATION_FAIL;
		this.message = CommonResult.AUTHENTICATION_FAIL.getMessage();
	}
	
	public NoAuthenticationException(Result result) {
		super(result);
		this.result = result;
		this.message = result.getMessage(MessageUtil::getMessage, result.getMessageId());
	}
	
	public NoAuthenticationException(Result result, String message) {
		super(result, message);
		this.result = result;
		this.message = message;
	}

	public NoAuthenticationException(Result result, Throwable cause) {
		super(result, cause);
		this.result = result;
		this.message = result.getMessage(MessageUtil::getMessage, result.getMessageId());
	}
	
    protected NoAuthenticationException(Result result, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
    	super(result, cause, enableSuppression, writableStackTrace);
    	this.result = result;
    	this.message = result.getMessage(MessageUtil::getMessage, result.getMessageId());
    }
}

File: modules/common/src/main/kotlin/com/media/bus/common/exceptions/NoAuthenticationException.kt
package com.media.bus.common.exceptions

import com.media.bus.common.result.Result
import com.media.bus.common.result.type.CommonResult

/** 인증에 실패할 경우 throw */
class NoAuthenticationException : BaseException {

    constructor() : super(CommonResult.AUTHENTICATION_FAIL)
    constructor(message: String) : super(CommonResult.AUTHENTICATION_FAIL, message)
    constructor(message: String, cause: Throwable) : super(CommonResult.AUTHENTICATION_FAIL, message, cause)
    constructor(cause: Throwable) : super(CommonResult.AUTHENTICATION_FAIL, cause)
    constructor(result: Result) : super(result)
    constructor(result: Result, message: String) : super(result, message)
    constructor(result: Result, cause: Throwable) : super(result, cause)
    protected constructor(result: Result, cause: Throwable, enableSuppression: Boolean, writableStackTrace: Boolean)
        : super(result, cause, enableSuppression, writableStackTrace)

    companion object {
        private const val serialVersionUID: Long = 6940889666952870454L
    }
}
